unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendRequestHUD/ReceivedFriendRequestHUDController.cs

```diff
@@ -0,0 +1,160 @@
+
+            view.OnClose += Hide;
+            view.OnOpenProfile += OpenProfile;
+            view.OnRejectFriendRequest += Reject;
+            view.OnConfirmFriendRequest += Confirm;
+            view.Close();
+
+            dataStore.HUDs.openReceivedFriendRequestDetail.OnChange += ShowOrHide;
+        }
+
+        public void Dispose()
+        {
+            dataStore.HUDs.openReceivedFriendRequestDetail.OnChange -= ShowOrHide;
+        }
+
+        private void ShowOrHide(string current, string previous)
+        {
+            if (string.IsNullOrEmpty(current))
+                Hide();
+            else
+                Show(current);
+        }
+
+        private void Show(string friendRequestId)
+        {
+            this.friendRequestId = friendRequestId;
+
+            FriendRequest friendRequest = friendsController.GetAllocatedFriendRequest(this.friendRequestId);
+
+            if (friendRequest == null)
+            {
+                Debug.LogError($"Cannot display friend request {friendRequestId}, is not allocated");
+                return;
+            }
+
+            view.SetBodyMessage(friendRequest.MessageBody);
+            view.SetTimestamp(DateTimeOffset.FromUnixTimeMilliseconds(friendRequest.Timestamp).DateTime);
+
+            var recipientProfile = userProfileBridge.Get(friendRequest.From);
+
+            if (recipientProfile != null)
+            {
+                view.SetSenderName(recipientProfile.userName);
+                view.SetSenderProfilePicture(recipientProfile.face256SnapshotURL);
+            }
+            else
+                Debug.LogError($"Cannot display user profile {friendRequest.From}, is not allocated");
+
+            var ownProfile = userProfileBridge.GetOwn();
+            view.SetOwnProfilePicture(ownProfile.face256SnapshotURL);
+
+            view.Show();
+        }
+
+        private void Hide()
+        {
+            dataStore.HUDs.openReceivedFriendRequestDetail.Set(null, false);
+            view.Close();
+        }
+
+        private void OpenProfile()
+        {
+            FriendRequest friendRequest = friendsController.GetAllocatedFriendRequest(friendRequestId);
+
+            if (friendRequest == null)
+            {
+                Debug.LogError($"Cannot open passport {friendRequestId}, is not allocated");
+                return;
+            }
+
+            openPassportVariable.Set(friendRequest.From);
+        }
+
+        private void Reject() =>
+            RejectAsync().Forget();
+
+        private async UniTaskVoid RejectAsync(CancellationToken cancellationToken = default)
+        {
+            view.SetState(ReceivedFriendRequestHUDModel.LayoutState.Pending);
+
+            try
+            {
+                await friendsController.RejectFriendshipAsync(friendRequestId)
+                                       .Timeout(TimeSpan.FromSeconds(10));
+                if (cancellationToken.IsCancellationRequested) return;
+
+                // TODO: send analytics
```

Comment:
If there is an issue to work on this, I would link it here with a `(#9999)` suffix after the TODO message

Reply:
done!